### PR TITLE
ext_job: writing arg specs to jobs.json

### PR DESCRIPTION
**Pull request by stefoss23**
**Task**
 ext_job: writing arg specs to jobs.json: The following needs to be written to jobs.json:
arg min
arg max
arg_types (int, str etc.)

**Approach**
Changes is ext_job_json_fprintf.

**Pre un-WIP checklist**
- [x] Statoil tests pass locally
- [x] Have completed graphical integration test steps

### CODE REVIEW

File: python/python/res/job_queue/ext_job.py [hunk 1]
```diff
@@ -143,17 +143,19 @@ def get_max_running_minutes(self):
...
         return self._min_arg()
-
-    def maximumArgumentCount(self):
+    
+    @property
+    def max_arg(self):
         return self._max_arg()
 
-    def argumentTypes(self):
+    def get_argument_types(self):
```
Comment (joakim-hove) [inline python/python/res/job_queue/ext_job.py]: Well;

if we are on the property bandwagon I suggest:
   
@property
def arg_types(self):
     ...

File: libjob_queue/src/ext_job.c [hunk 2]
```diff
@@ -743,6 +748,40 @@ static void __fprintf_python_argList(FILE * stream,
...
+                                       const char * prefix,
+                                       const char * key,
+                                       const ext_job_type * ext_job,
+                                       const char * suffix,
+                                       const char * null_value) {
+  fprintf(stream, "%s", prefix);
+  if (ext_job->arg_types) {
+    fprintf(stream, "\"%s\" : [", key);
+    int i;
+    for (i = 0; i < ext_job->max_arg; i++) {
```
Comment (joakim-hove) [inline libjob_queue/src/ext_job.c:761]: This should loop over the size of arg_types - which can very well be less than max_arg.

File: libjob_queue/src/ext_job.c [hunk 3]
```diff
@@ -743,6 +748,40 @@ static void __fprintf_python_argList(FILE * stream,
...
+      switch(type) {
+        case CONFIG_INT:    arg_type = JOB_INT_TYPE; break;
+        case CONFIG_FLOAT:  arg_type = JOB_FLOAT_TYPE; break;
+        case CONFIG_STRING: arg_type = JOB_STRING_TYPE; break;
+        case CONFIG_BOOL:   arg_type = JOB_BOOL_TYPE; break;
+        default: util_abort("ERROR in %s", __func__);
+      }
+
+      fprintf(stream, "\"%s\"", arg_type);      
+      if ((i + 1) < ext_job->max_arg)
```
Comment (joakim-hove) [inline libjob_queue/src/ext_job.c:774]: Again - consult length of arg_types.

**Review by joakim-hove — APPROVED** [review]
Very good - some minor comments; then she goes in!

File: libjob_queue/src/ext_job.c (hunk 2 above)
Comment (stefoss23) [reply libjob_queue/src/ext_job.c:761]: I am not sure about that. arg_types should be set in the forward model config file and cannot be changed by the user (from the ert config file).

File: libjob_queue/src/ext_job.c (hunk 3 above)
Comment (stefoss23) [reply libjob_queue/src/ext_job.c:774]: As above - unsure.

File: libjob_queue/src/ext_job.c (hunk 2 above)
Comment (joakim-hove) [reply libjob_queue/src/ext_job.c:761]: The point is that you do not have to specify type for all the arguments, i.e. you can have the following situation:
```txt
MIN_ARG 1
MAX_ARG 3
ARG_TYPES 0 INT
```
In this case the function should have 1-3 arguments, the first argument should be an integer - *you do not care* about the two next arguments - meaning that they will be `CONFIG_STRING`.

I think the best solution is to use your implementation, but replace `int_vector_iget`with `int_vector_safe_iget()` which will return the default value - i.e. `CONFIG_STRING` if you ask for an out-of-range value.

**Pull request merged**